File: tests/system/fixtures/http-listener.py
import sys
from http.server import BaseHTTPRequestHandler, HTTPServer


port = int(sys.argv[1])
marker = sys.argv[2] if len(sys.argv) > 2 else "http-listener"


class Handler(BaseHTTPRequestHandler):
    def do_GET(self):
        body = marker.encode("utf-8")
        self.send_response(200)
        self.send_header("content-type", "text/plain; charset=utf-8")
        self.send_header("content-length", str(len(body)))
        self.end_headers()
        self.wfile.write(body)

    def log_message(self, format, *args):
        return


HTTPServer(("127.0.0.1", port), Handler).serve_forever()

File: packages/sandboxd/tests/fixtures/http-listener.js
import http from "node:http";

const port = Number(process.argv[2]);
const marker = process.argv[3] ?? "http-listener";

if (!Number.isInteger(port) || port <= 0) {
  throw new Error(`Expected a positive integer port, got ${String(process.argv[2])}`);
}

const server = http.createServer((_request, response) => {
  response.writeHead(200, {
    "content-type": "text/plain; charset=utf-8",
  });
  response.end(`ok:${marker}`);
});

server.listen(port, "127.0.0.1");

function shutdown() {
  server.close(() => {
    process.exit(0);
  });
}

process.on("SIGINT", shutdown);
process.on("SIGTERM", shutdown);
